### PR TITLE
feat: continue refinment of compose builder for zero trust

### PR DESCRIPTION
**Pull request by dovholuknf**
Continuing towards a compose builder that will emit a zero trust edgex configuration. 

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] I am not introducing a breaking change (if you are, flag in conventional commit message with `BREAKING CHANGE:` describing the break)
- [x] I have fully tested (add details below) this the new feature or bug fix (if not, why?)
- [ ] I have opened a PR for the related docs change (if not, why?)
  <link to docs PR>

### CODE REVIEW

File: compose-builder/backup/docker-compose.yml.config [hunk 1]
```diff
@@ -0,0 +1,1365 @@
+name: edgex
```
Comment (lenny-intel) [inline compose-builder/backup/docker-compose.yml.config:1]: What is this file for? Temp file?

File: compose-builder/docker-compose-base.yml [hunk 2]
```diff
@@ -35,8 +35,6 @@ services:
     image: hashicorp/consul:${CONSUL_VERSION}
     command: "agent -ui -bootstrap -server -client 0.0.0.0"
     user: "root:root" # Note that Consul is run under the 'consul' user, but entry point scripts need to first run as root
-    ports:
```
Comment (lenny-intel) [inline compose-builder/docker-compose-base.yml:38]: With your new discovery these can stay and add `ports: !reset null` to the zero-trust yaml?

File: compose-builder/gen_secure_compose_ext.sh [hunk 3]
```diff
@@ -47,10 +47,22 @@ mkdir -p "$GEN_EXT_DIR"
...
+if [ "$ZERO_TRUST" = "1" ]; then
+  sed -i 's,${ZERO_TRUST},#,g' "$SERVICE_EXT_COMPOSE_PATH"
+  cat >> "$SERVICE_EXT_COMPOSE_PATH" <<HERE
+    # env_file does not override environment and these values are set in the add-* templates
+    # use a heredoc and append it to the generated file accordingly so that docker compose will
+    # reduce it down and override as expected
+    environment:
+      SERVICE_HOST: ${service_name}.edgex.ziti
+      SERVICE_PORT: 80
+    ports: !reset null
```
Comment (lenny-intel) [inline compose-builder/gen_secure_compose_ext.sh:63]: NICE!

File: compose-builder/add-docker-compose-base-ports.yml [hunk 4]
```diff
@@ -0,0 +1,40 @@
+services:
```
Comment (lenny-intel) [inline compose-builder/add-docker-compose-base-ports.yml:1]: This  file can now be removed with your new discovery of `ports: !reset null`

File: compose-builder/docker-compose-base-zero-trust.yml [hunk 5]
```diff
@@ -0,0 +1,91 @@
+name: edgex
+services:
+  app-rules-engine:
+    environment:
+      SERVICE_HOST: app-rules-engine.edgex.ziti
+      SERVICE_PORT: 80
+  core-command:
+    environment:
```
Comment (lenny-intel) [inline compose-builder/docker-compose-base-zero-trust.yml:8]: Do this for all the services.
```suggestion
    ports: !reset null
    environment:
```

File: compose-builder/Makefile [hunk 6]
```diff
@@ -974,6 +985,15 @@ define COMPOSE_DOWN
 		down $1
 endef
 
+ifeq (zero-trust, $(filter zero-trust,$(ARGS)))
+	# overrides for zero trust from the base files
+	COMPOSE_FILES:=$(COMPOSE_FILES) -f docker-compose-base-zero-trust.yml
```
Comment (lenny-intel) [inline compose-builder/Makefile:990]: Rename file since it is not a base, but an addition...
```suggestion
	COMPOSE_FILES:=$(COMPOSE_FILES) -f add-security-zero-trust.yml
```

File: compose-builder/Makefile [hunk 7]
```diff
@@ -974,6 +985,15 @@ define COMPOSE_DOWN
...
+else
+	COMPOSE_FILES:=$(COMPOSE_FILES) -f add-docker-compose-base-ports.yml
```
Comment (lenny-intel) [inline compose-builder/Makefile:992]: Not needed with `ports: !reset null`
```suggestion
```

**Review by lenny-intel — CHANGES_REQUESTED** [review]

File: compose-builder/docker-compose-base.yml (hunk 2 above)
Comment (dovholuknf) [reply compose-builder/docker-compose-base.yml]: yep. pushed that change up

File: compose-builder/Makefile (hunk 6 above)
Comment (dovholuknf) [reply compose-builder/Makefile]: all set

File: compose-builder/Makefile [hunk 8]
```diff
@@ -1034,8 +1052,11 @@ pull: gen
 
 gen:
 	echo MQTT_VERBOSE=${MQTT_VERBOSE}
-	${DOCKER_COMPOSE}  -p edgex $(COMPOSE_FILES) ${GEN_COMMAND} > docker-compose.yml
-	rm -rf ./$(GEN_EXT_DIR)
+	${DOCKER_COMPOSE}  -p edgex $(COMPOSE_FILES) ${GEN_COMMAND} > docker-compose.yml.gen
+	if [ "$(NO_CLEANUP)" = "0" ]; then rm -rf ./$(GEN_EXT_DIR); fi
+	echo "# Generated with: $(docker_ver)" > docker-compose.yml
```
Comment (lenny-intel) [inline compose-builder/Makefile:1057]: Doesn't this overwrite the file with just the version? Probably need to be moved above the gen and switch the gen to be a >>
```suggestion
	echo "# Generated with: $(docker_ver)" > docker-compose.yml
```

**Review by lenny-intel — CHANGES_REQUESTED** [review]

File: compose-builder/Makefile (hunk 8 above)
Comment (dovholuknf) [reply compose-builder/Makefile:1057]: yes but then it gets appended to below from the generated file, and then that generated file is removed, resulting in a file with a 'header' of what docker version generated it and the rest of the contents in this file:
```
	echo "# Generated with: $(docker_ver)" > docker-compose.yml
	cat docker-compose.yml.gen >> docker-compose.yml
	rm docker-compose.yml.gen
```

the second line cats the generated file into the resultant

Comment (dovholuknf) [reply compose-builder/Makefile:1057]: oh i see what you're thinknig there... sure i can adapt

File: compose-builder/Makefile [hunk 9]
```diff
@@ -1034,8 +1052,9 @@ pull: gen
 
 gen:
 	echo MQTT_VERBOSE=${MQTT_VERBOSE}
-	${DOCKER_COMPOSE}  -p edgex $(COMPOSE_FILES) ${GEN_COMMAND} > docker-compose.yml
-	rm -rf ./$(GEN_EXT_DIR)
+	echo "# Generated with: $(docker_ver)" > docker-compose.yml
+	${DOCKER_COMPOSE}  -p edgex $(COMPOSE_FILES) ${GEN_COMMAND} >> docker-compose.yml
+	if [ "$(NO_CLEANUP)" = "0" ]; then rm -rf ./$(GEN_EXT_DIR); fi
```
Comment (lenny-intel) [inline compose-builder/Makefile:1057]: Actually this is out of scope for this PR and will cause updates to all the generated files.

I like it, but it should be in a separate PR.

Comment (dovholuknf) [reply compose-builder/Makefile]: ok. i reverted the docker version echo

**Review by lenny-intel — APPROVED** [review]
LGTM pending more testing

**Review by lenny-intel — CHANGES_REQUESTED** [review]

**Review by lenny-intel — APPROVED** [review]
LGTM
